### PR TITLE
SOLR-9775 fixed NPEs

### PR DESCRIPTION
**Pull request by kagan770**

### CODE REVIEW

**Comment by cpoerschke** [comment]
Hi Roman, thanks for your pull request. I have created https://issues.apache.org/jira/browse/SOLR-9775 to correspond to it.

File: solr/core/src/java/org/apache/solr/search/QueryResultKey.java [hunk 1]
```diff
@@ -49,12 +49,14 @@ public QueryResultKey(Query query, List<Query> filters, Sort sort, int nc_flags)
...
-        h += filt.hashCode();
+        if (filt != null)
+          h += filt.hashCode();
     }
 
     sfields = (this.sort !=null) ? this.sort.getSort() : defaultSort;
     for (SortField sf : sfields) {
-      h = h*29 + sf.hashCode();
+      if (sf != null)
+        h = h*29 + sf.hashCode();
```
Comment (risdenk) [inline solr/core/src/java/org/apache/solr/search/QueryResultKey.java:59]: add `{}` around for if statement

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/core/src/java/org/apache/solr/search/QueryResultKey.java [hunk 2]
```diff
@@ -49,12 +49,14 @@ public QueryResultKey(Query query, List<Query> filters, Sort sort, int nc_flags)
       for (Query filt : filters)
         // NOTE: simple summation used here so keys with the same filters but in
         // different orders get the same hashCode
-        h += filt.hashCode();
+        if (filt != null)
+          h += filt.hashCode();
```
Comment (risdenk) [inline solr/core/src/java/org/apache/solr/search/QueryResultKey.java:53]: add `{}` around for if statement

**Comment by epugh** [comment]
See https://github.com/apache/solr/pull/1107

**Pull request closed without merging**